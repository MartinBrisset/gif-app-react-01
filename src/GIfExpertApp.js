import React, {useState} from 'react'
import { AddCategoria } from './components/AddCategoria'
import { GifGrid } from './components/GifGrid'

export const GIfExpertApp = ({ defaultCategorias = []}) => {

    // const [categorias, setCategorias] = useState(['One Punch'])
    const [categorias, setCategorias] = useState(defaultCategorias)

    return (
        <div>
            <h2>GIfExpertApp</h2>
            <AddCategoria setCategorias={setCategorias}/>
            <hr  />

            <ol>
                {
                    categorias.map( categoria => {
                        return <GifGrid 
                            key={categoria}
                            categoria={categoria} 
                        />
                    })
                }
            </ol>

        </div>
    )
}
