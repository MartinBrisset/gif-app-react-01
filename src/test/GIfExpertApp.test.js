import React from "react"
import { shallow } from "enzyme"
import { GIfExpertApp } from "../GIfExpertApp"


describe('Pruebas en componente GIfExpertApp', () => {
    
    test('Debe mostrar correctamente', () => {
        
        const wrapper = shallow(<GIfExpertApp  />)
        expect( wrapper ).toMatchSnapshot()

    })

    test('debe de mostrar una lista de categorias ', () => {
        
        const categorias = ['Goku', 'Vegeta']
        const wrapper = shallow(<GIfExpertApp defaultCategorias={ categorias}  />)

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GifGrid').length ).toBe( categorias.length )

    })
    
    

})
