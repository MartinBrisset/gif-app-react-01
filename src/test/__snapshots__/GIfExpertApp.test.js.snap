// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GIfExpertApp Debe mostrar correctamente 1`] = `
<div>
  <h2>
    GIfExpertApp
  </h2>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas en componente GIfExpertApp debe de mostrar una lista de categorias  1`] = `
<div>
  <h2>
    GIfExpertApp
  </h2>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Goku"
      key="Goku"
    />
    <GifGrid
      categoria="Vegeta"
      key="Vegeta"
    />
  </ol>
</div>
`;
